src/type_checker.cpp

```diff
@@ -140,6 +140,51 @@ void TypeChecker::Visit(ContinueStmtNode& continue_stmt) {
   }
 }
 
+namespace {
+/// @brief A shared state to convey the presence of a default label in a switch
+/// statement.
+/// @note To allow nested switch statements, the state is stacked.
+auto
+    switch_already_has_default  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
```

Comment:
I think `switch_has_default` should be clear enough.

Reply:
I believe that the word `already` better conveys that the default shouldn't appear twice.
Consider the use case:

```c++
if (switch_has_default.back()) {
  // error
}
```

and

```c++
if (switch_already_has_default.back()) {
  // error
}
```